### PR TITLE
testing: add test actions to editor context menus

### LINKED ISSUE #130548: Expose Run and Debug test to other UI entries?
Hi @connor4312,

Bellow are some rough ideas came up in my mind, would like to hear your thoughts on them :)

The new testing API provide functionality like `Run/Debug tests in current file`, `Run/Debug tests at cursor`. Since now they can only be triggered from the Command Palette, which users might even do not know their existence. I'm thinking if it's possible to promote them in different UI entries?

- Put `Run/Debug tests in current file` to `editor/title/run` when current file has tests? 
  - There is a separate discussion in https://github.com/microsoft/vscode-java-debug/issues/1037
- Put `Run/Debug tests at cursor` in to the context menu, like when debugging we have `run to cursor`:
  image

### PR DESCRIPTION
**Pull request by connor4312**
Fixes #130548. Adds an efficient test-by-uri lookup that are used by decorations and also to set a new context key.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
 	"search.exclude": {
 		"**/node_modules": true,
 		"**/bower_components": true,
+		"cli/target/**": true,
```
Comment (connor4312) [inline .vscode/settings.json:21]: small unrelated fix: I noticed search was getting slow, which I assumed was Defender, but it was actually looking through all the build output in the CLI

File: src/vs/workbench/contrib/testing/common/testTypes.ts [hunk 2]
```diff
@@ -687,80 +687,17 @@ export abstract class AbstractIncrementalTestCollection<T extends IncrementalTes
...
-						parent.children.add(internalTest.item.extId);
-						const created = this.createItem(internalTest, parent);
-						this.items.set(internalTest.item.extId, created);
-						changes.add(created);
-					}
-
-					if (internalTest.expand === TestItemExpandState.BusyExpanding) {
-						this.busyControllerCount++;
-					}
+				case TestDiffOpType.Add:
```
Comment (connor4312) [inline src/vs/workbench/contrib/testing/common/testTypes.ts:690]: The logic in these cases were moved down verbatim. I was originally considering overriding them and didn't, but I think this is still some decent cleanup

**Review by lramos15 — APPROVED** [review]
